### DISCUSSION TITLE
How to find the referenced note

### DISCUSSION THREAD

**Discussion by abdcef** [brianpetro/obsidian-smart-connections#127] [Q&A] [answered]
Hello again Bryan,

I am sharing an example here as I am unable to figure out how I can quickly get to the source note from the chat.

![CleanShot 2023-03-27 at 22 33 46@2x](https://user-images.githubusercontent.com/56148250/227972300-c1d4f42c-a571-4404-885d-eb49294f9bc9.png)

What is note #10, and how can I quickly go to it? It was second in the context code block.

**Comment by brianpetro** [comment, marked as answer]
@abdcef I'm still working on this. The current UIUX is a bit lacking, but it can be done: 

Please look at the three buttons in the bottom right of the response. The middle button will copy the context used by the AI in that response to your clipboard.

You can then review the context and should be able to figure out the correct source from there.

In the near future, the links directly to the notes used as context will also be available.

Similarly, the "eye" icon button will copy a code block to your clipboard. This will render a list of notes used as context, _but_, this list is not currently reliable as the exact source of context used. This is because the rendering still uses the original calculation for relevant context, while the Smart Chat uses a weighted calculation for more efficient use of the context window. 

There are still many moving parts that seeing an accurate view of the context depends on, but the feature you're seeking is definitely on the road map. I hope the above explanation helps in the meantime!

Brian🌴
